soroban-auth/src/lib.rs

```diff
@@ -11,6 +11,7 @@ pub use crate::public_types::{
 fn check_ed25519_auth(env: &Env, auth: &Ed25519Signature, function: Symbol, args: Vec<RawVal>) {
     let msg = SignaturePayloadV0 {
         function,
+        call_cntxt: env.get_parent_call_stack(),
```

Comment:
If we're going to move ahead with this we need a way to make the domain generic so that a developer can choose between using the complete stack, or just a single contract.

Maybe this could be separate SignaturePayload values?

Reply:
Yeah there isn't much flexibility here. The developer is forced to take the call stack into account event if they don't want to. I'm going to try making the domain generic to see what the options are.